src/algod/v2/mod.rs

```diff
@@ -238,3 +254,54 @@ impl Algod {
         Ok(self.client.versions().await?)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_valid_client_builder() {
+        let res = Algod::new(
+            "http://example.com",
+            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        );
+        assert!(res.ok().is_some());
+    }
+
+    #[test]
+    fn test_client_builder_with_invalid_url() {
+        let res = Algod::new(
+            "asfdsdfs",
+            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        );
+        assert!(res.is_err());
+        assert!(matches!(res.err().unwrap(), AlgonautError::BadUrl(_)));
+    }
+
+    #[test]
+    fn test_client_builder_with_invalid_url_no_scheme() {
+        let res = Algod::new(
+            "example.com",
+            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        );
+        assert!(res.is_err());
+        assert!(matches!(res.err().unwrap(), AlgonautError::BadUrl(_)));
+    }
+
+    #[test]
+    fn test_client_builder_with_invalid_token() {
+        let res = Algod::new(
+            "http://example.com",
+            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        );
+        assert!(res.is_err());
+        assert!(res.err().unwrap() == AlgonautError::BadToken);
+    }
+
+    #[test]
```

Comment:
Does this test make sense? Is it allowed to have an empty token when connecting, for instance, to AlgoExplorer's node?

Reply:
In this case you'd use the `with_headers` initializer and don't pass the token header at all https://github.com/manuelmauro/algonaut/blob/8ba4a464787b8cbdb6aba7e9eefcf1e40c3021eb/src/algod/v2/mod.rs#L29